proposals/0282-record-dot-syntax.rst

```diff
@@ -301,6 +363,50 @@ repository <https://github.com/ndmitchell/record-dot-preprocessor>`__.
 Those tests include infix applications, polymorphic data types,
 interoperation with other extensions and more.
 
+
+Reserved keywords and other special field names
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The inclusion of *reservedid* in *field* means that the following is accepted:
+
+.. code:: haskell
+
+   data Foo = Foo { fooType :: FooType }
+
+   instance HasField "type" Foo FooType where
+     getField = fooType
+
+   instance SetField "type" Foo FooType where
+     setField t foo = foo { fooType = t }
+
+   e :: Foo -> FooType
+   e foo = foo.type            -- Translates to getField @"type" foo
+
+   f :: FooType -> Foo -> Foo
+   f t foo = foo { type = t }  -- Translates to setField @"type" t foo
+
+However the following continue to be rejected:
+
+.. code:: haskell
+
+   data Foo = Foo { type :: FooType }  -- Error: record datatype field cannot be reserved word
+
+   x foo = foo.TYPE     -- Error: field must start with a lowercase letter
+
+   y foo = foo."type"  -- Error: field cannot be double-quoted
+
+The latter two are slightly inconsistent with ``OverloadedLabels``, which
+permits ``#TYPE`` and ``#"type"`` as labels.
```

Comment:
Is this accurate? Above it says that `foo.TYPE` and `foo."type"` should trigger name resolution errors rather than syntax errors such as "field cannot be double-quoted". For example, I would expect `data Foo = Foo { bar :: Int }; x foo = foo."bar"` to work despite `"bar"` being double-quoted.

Reply:
Where precisely are you referring to?

I believe the PR is internally consistent in that it does *not* currently permit double-quoted or uppercase identifiers following a record dot. If the committee votes to allow them (which is my preferred option), I'll amend it and ask for a final check before merging.

Reply:
You're right; the proposal as written seems to be consistent. By "Above it says" I was referring to

> (While the changes to *fbind* allow more record constructions to parse, a construction such as ``C { type = e }}`` or ``C { foo.bar = e }`` will continue to be rejected during name resolution.)

However, while the currently proposed syntax for `*field*` does allow `foo.type`, it does not allow `foo."type"`. Apologies.